pkg/relayer/proxy/proxy.go

```diff
@@ -59,60 +69,79 @@ func NewRelayerProxy(
 	clientCtx sdkclient.Context,
 	keyName string,
 	keyring keyring.Keyring,
-
+	nativeServicesListenAddress map[serviceId]string,
 	// TODO_INCOMPLETE(@red-0ne): Uncomment once the BlockClient interface is available.
 	// blockClient blocktypes.BlockClient,
 ) RelayerProxy {
 	accountQuerier := accounttypes.NewQueryClient(clientCtx)
 	supplierQuerier := suppliertypes.NewQueryClient(clientCtx)
 	sessionQuerier := sessiontypes.NewQueryClient(clientCtx)
-	providedServices := buildProvidedServices(ctx, supplierQuerier)
 	servedRelays, servedRelaysProducer := channel.NewObservable[*types.Relay]()
 
 	return &relayerProxy{
 		// TODO_INCOMPLETE(@red-0ne): Uncomment once the BlockClient interface is available.
-		// blockClient:          blockClient,
-		keyName:              keyName,
-		keyring:              keyring,
-		accountsQuerier:      accountQuerier,
-		supplierQuerier:      supplierQuerier,
-		sessionQuerier:       sessionQuerier,
-		providedServices:     providedServices,
-		servedRelays:         servedRelays,
-		servedRelaysProducer: servedRelaysProducer,
+		// blockClient:       blockClient,
+		keyName:                     keyName,
+		keyring:                     keyring,
+		accountsQuerier:             accountQuerier,
+		supplierQuerier:             supplierQuerier,
+		sessionQuerier:              sessionQuerier,
+		nativeServicesListenAddress: nativeServicesListenAddress,
+		servedRelays:                servedRelays,
+		servedRelaysProducer:        servedRelaysProducer,
 	}
 }
 
-// Start starts all supported proxies and returns an error if any of them fail to start.
+// Start concurrently starts all advertised relay servers and returns an error if any of them fails to start.
 func (rp *relayerProxy) Start(ctx context.Context) error {
-	panic("TODO: implement relayerProxy.Start")
+	// The provided services map is built from the supplier's on-chain advertised information,
+	// which is a runtime parameter that can be changed by the supplier.
+	// Build the provided services map at Start instead of NewRelayerProxy to avoid having to
+	// return an error from the constructor.
+	err := rp.BuildProvidedServices(ctx)
+	if err != nil {
+		return err
+	}
+
+	startGroup, gctx := errgroup.WithContext(ctx)
```

Comment:
I was looking in https://pkg.go.dev/golang.org/x/sync/errgroup and they seem to reuse the `ctx` varname.

Consider either renaming to `relayServerCtx` or just overriding `ctx`. Ditto below.

```
func MD5All(ctx context.Context, root string) (map[string][md5.Size]byte, error) {
	// ctx is canceled when g.Wait() returns. When this version of MD5All returns
	// - even in case of error! - we know that all of the goroutines have finished
	// and the memory they were using can be garbage-collected.
	g, ctx := errgroup.WithContext(ctx)
```